### ISSUE TITLE
Buffs Stack (BB #718)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#718]
This issue was migrated from bitbucket.
**Original Reporter:** Forthehorde
**Original Date:** 13.10.2014 20:35:39 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718

<hr>

Buffs are stackable.
on the pic is ID-27502 - [Scroll of Stamina V] but I think every buffs can stack

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Forthehorde
**Original Date:** 13.10.2014 20:38:00 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12943839

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 13.10.2014 21:20:59 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12944555

<hr>

Cheers, if you'd like to assist further I could use a copy of each scrolls item or spell ID. Spell pref.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 13.10.2014 21:50:56 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12944992

<hr>

```
                // Strength
                case 8118:
                // Stamina
                case 8099:
                // Spirit
                case 8112:
                //Intellect
                case 8096:
                // Agility
                case 8115:
                // Armor
                case 8091:
```

These should be the first ranks, won't take me long to fix this.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 13.10.2014 23:50:41 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12946395

<hr>

Hmm, can A scroll of Strength stack with a scroll of stamina and that be blzzlike?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 14.10.2014 04:59:04 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12949450

<hr>

```
#!sql

DELETE FROM spell_group WHERE id IN (1021);
INSERT INTO spell_group (id, spell_id) VALUES
-- Add missing Paladin seal to stack rules
(1017, 20375), -- Seal of Command
-- Add stack groups for scroll spells
(1021, 8118),  -- Scroll of Strength
(1021, 8099),  -- Scroll of Stamina
(1021, 8112),  -- Scroll of Spirit
(1021, 8096),  -- Scroll of Intellect
(1021, 8115),  -- Scroll of Agility
(1021, 8091);  -- Scroll of Armor

DELETE FROM spell_group_stack_rules WHERE group_id = 1021;
INSERT INTO spell_group_stack_rules (group_id, stack_rule) VALUES
(1021, 1);
```

Give this a try :)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Forthehorde
**Original Date:** 14.10.2014 05:10:40 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12949578

<hr>

I can buff with Scroll of Stamina like 50 times check the pic,
same for every exist buffs in game

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 14.10.2014 05:19:19 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12949670

<hr>

bug not confirmed on last rev

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Forthehorde
**Original Date:** 14.10.2014 05:26:19 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12949757

<hr>

Add 27502 just 1 to yourself buff up relog and then add agian 27502 (`1) buff up agian.
If you have more than 1 buff in stack you cant see the bug,you need to separate scroll only 1

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 14.10.2014 05:44:40 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12949937

<hr>

confirmed, with scrolls, elixirs and potions

use scroll(potion, elixir) + logoff + logon + use again -> profit

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Forthehorde
**Original Date:** 14.10.2014 05:54:42 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12950032

<hr>

Everything can be stacked.
Good luck with the fixes.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 14.10.2014 06:03:14 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12950143

<hr>

not everything

class casts can not be stacked

tried Power Word Fortitude and Prayer of Shadow Protection

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 14.10.2014 06:32:14 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12950556

<hr>

Problem is only with item spells (scrolls, potions, elixirs)

spell_group useless

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 14.10.2014 09:46:16 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12954642

<hr>

This is the source of the problem :)

https://bitbucket.org/oregon/oregoncore/commits/6f604193227e9ec6dee4b58262e79d1f6e5ddb98

delete this commit please

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 14.10.2014 12:02:42 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12957513

<hr>

Another way to fix it:

if (Aur->GetCastItemGUID() && !aurSpellInfo->StackAmount && **Aur->GetId() == 28093**)

tested - works fine for me

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Forthehorde
**Original Date:** 14.10.2014 15:29:49 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12962935

<hr>

if (Aur->GetCastItemGUID() && !aurSpellInfo->StackAmount && Aur->GetId() == 28093)
Works fine.
Thanks

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** bobkhan
**Original Date:** 14.10.2014 18:52:57 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12967004

<hr>

fix?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 14.10.2014 18:55:02 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-12967041

<hr>

this is fix

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 22.10.2014 00:27:29 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-13130254

<hr>

[DB/SpellGroups] Prevent Scrolls from stacking with one another. Add Seal of Command
                 to Seal stack group.

```
             Fix #718.
```

→ <<cset 735291a17625>>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 22.10.2014 15:45:23 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-13146700

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 22.10.2014 15:59:14 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-13147030

<hr>

Use the fix you provided until Zaffy looks at it, chances are we'll merge your fix into the core.

You don't even need to log out and in, you just use two different stacks of scrolls.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 22.10.2014 17:01:40 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-13148325

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 25.10.2014 16:30:35 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/718/#comment-13208427

<hr>

Fix issue #718

Sorry for the delay... I'm quite busy right now.
The commit that caused this commit was on the good way,
but still made some troubles around with aura stacking.

Auras should stack even if they are not stackable but casted
from different enchanted items/weapons. So for now, let's
keep the mongoose stacking hackfixed. The others spells
will follow when we'll know how to better solve that issue.

→ <<cset 7cef6882ed08>>
